lab_4_summarization_textrank/README.md

```diff
@@ -0,0 +1,758 @@
+
+У каждого объекта `TextRankSummarizer` **обязательно** должны быть следующие атрибуты:
+* `_graph` - здесь хранится пришедший на вход заполненная матрица, которая, как мы помним, является репрезентацией текста
+* `_damping_factor` - как упоминалось ранее, это константное значение, нужное для того, чтобы алгоритм не зациклился и отражающее вероятность перейти от конкретной вершины к любой другой; установите значение данного атрибута как 0.85
+* `_convergence_threshold` - это константное значение, отражающее максимально допустимую разницу между весами до и после обновления: если после очередной итерации пересчета весов вершин сумма разностей вершин не превышает этот порог, то мы считаем, что оптимальные веса найдены; установите значение данного атрибута как 0.0001
+* `_max_iter` - это константное значение, отражающее максимально допустимое количество итераций обновления весов: теоретически на вход может прийти очень сложный граф, в котором невозможно подобрать стабильные веса, и чтобы не попасть в бесконечный цикл, мы ограничиваем максимальное количество итераций; установите значение данного атрибута как 50
+* `_scores` - веса вершин, здесь ключами являются вершины (предложения), а значениями их вес, отражающий их важность
+
+Интерфейс:
+```py
+class TextRankSummarizer:
+    def __init__(self, graph: SimilarityMatrix) -> None:
+        pass
+```
+
+#### Шаг 7.2 Обновление веса конкретной вершины
+
+Модифицируйте обновление веса конкретной вершины. Метод принимает на вход вершину, вес которой необходимо пересчитать, 
+список похожих на неё вершин и словарь с весами вершин после предыдущей итерации обновления. 
+
+Новый вес вершины необходимо рассчитать по формуле из Шага №5 (то есть модифицировать уже существующий в классе `VanillaTextRank` метод расчёта). 
+Для получения информации о значении $d$ необходимо обратиться к соответствующему атрибуту класса, 
+информацию о количестве похожих вершин нужно взять при помощи метода матрицы похожести `calculate_inout_score`. Доступ к матрице также происходит через атрибуты класса.
+
+Метод ничего не возвращает. Вместо этого он перезаписывает вес вершины в словаре, хранящемся в атрибуте `_scores`. 
+
+Интерфейс: 
+```py
+def update_vertex_score(self, 
+                        vertex: Sentence,
+                        incidental_vertices: list[Sentence],
+                        scores: dict[Sentence, float]) -> None:
+    pass
+```
+
+#### Шаг 7.3 Итеративное обновление весов вершин
+
+Модифицируйте метод итеративного переподсчёта важности предложений `train`. 
+
+Метод ничего не принимает на вход и ничего не возвращает. Вместо этого он изменяет атрибут `_scores`.
+
+#### Шаг 7.4 Получение наиболее важных предложений
+
+Пора определять метод, возвращающий наиболее важные предложения. 
+
+Метод принимает на вход аргумент, обозначающий требуемое количество предложений. Метод возвращает последовательность предложений с самым высоким весом.
```

Comment:
make it more clear not a list of sentences with the biggest weight, but top-N of sentences ordered in decreasing order based on their importance.